From: Chen Yu <yu.c.chen@intel.com>
Date: Thu, 2 Apr 2020 15:56:52 +0800
Subject: PM: sleep: Add pm_debug_messages kernel command line option
Git-commit: db96a75946d3a2a41092c83992e65f727cd35529
Patch-mainline: v5.7-rc1
References: bsc#1186752

Debug messages from the system suspend/hibernation infrastructure
are disabled by default, and can only be enabled after the system
has boot up via /sys/power/pm_debug_messages.

This makes the hibernation resume hard to track as it involves system
boot up across hibernation.  There's no chance for software_resume()
to track the resume process, for example.

Add a kernel command line option to set pm_debug_messages during
boot up.

Signed-off-by: Chen Yu <yu.c.chen@intel.com>
[ rjw: Subject & changelog ]
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Jean Delvare <jdelvare@suse.de>
---
 Documentation/admin-guide/kernel-parameters.txt |    3 +++
 kernel/power/main.c                             |    7 +++++++
 2 files changed, 10 insertions(+)

--- a/Documentation/admin-guide/kernel-parameters.txt
+++ b/Documentation/admin-guide/kernel-parameters.txt
@@ -3610,6 +3610,9 @@
 			Override pmtimer IOPort with a hex value.
 			e.g. pmtmr=0x508
 
+	pm_debug_messages	[SUSPEND,KNL]
+			Enable suspend/resume debug messages during boot up.
+
 	pnp.debug=1	[PNP]
 			Enable PNP debug messages (depends on the
 			CONFIG_PNP_DEBUG_MESSAGES option).  Change at run-time
--- a/kernel/power/main.c
+++ b/kernel/power/main.c
@@ -417,6 +417,13 @@ static ssize_t pm_debug_messages_store(s
 
 power_attr(pm_debug_messages);
 
+static int __init pm_debug_messages_setup(char *str)
+{
+	pm_debug_messages_on = true;
+	return 1;
+}
+__setup("pm_debug_messages", pm_debug_messages_setup);
+
 /**
  * __pm_pr_dbg - Print a suspend debug message to the kernel log.
  * @defer: Whether or not to use printk_deferred() to print the message.
